lib/music/MusicInterface.js

```diff
@@ -0,0 +1,219 @@
+	 * Returns the guilds current player
+     * @since 2.0.1
+	 * @returns {Object<Player>}
+	 * @readonly
+	 */
+    get player() {
+        return this.client.lavalink.get(this.guild.id) || null;
+    }
+
+    /**
+	 * The method to add a LavalinkTrack to the queue
+     * @since 2.0.1
+	 * @param {LavalinkTrack} songdata The LavalinkTrack data to be added and formated for the queue to handle
+	 * @param {Discord.GuildMember} requester The person who requested the addition of this song
+	 * @returns {Song}
+	 */
+    add(songdata, requester) {
+        const song = new Song(songdata, requester);
+        this.queue.push(song);
+        return song;
+    }
+
+    /**
+	 * Makes the bot join a given voice channel with the optimal region and into the specified voiceChannel
+     * @since 2.0.1
+	 * @param {Discord.voiceChannel} voiceChannel The voice channel the player should join
+	 * @returns {Object<Player>}
+	 */
+    join(voiceChannel) {
+        return this.client.lavalink.join({
+            guild: this.guild.id,
+            channel: voiceChannel.id,
+            host: this.client.lavalink.getIdealHost(this.guild.region)
+        }, { selfdeaf: this.guild.configs.selfDeaf ? this.guild.configs.selfDeaf : true });
+    }
+
+    /**
+	 * Makes the bot leave the voiceChannel it is currently in and set the playing status to false
+     * @since 2.0.1
+	 * @param {boolean} [force=true] If the bot should force leave the curreny channel
+	 * @returns {this}
+	 */
+    async leave(force = true) {
+        if (this.playing || this.player || force) this.player.stop();
+        await this.client.lavalink.leave(this.guild.id);
+        this.playing = false;
+        return this;
+    }
+
+    /**
+	 * Starts the player and checks if all criteria is met otherwise it will throw the warning but if it works it will start the player
+     * @since 2.0.1
+	 * @returns {Player}
+	 */
+    async play() {
+        const check = !this.voiceChannel ? "Well, Well, Well, It seems theres no where to play the sick tunes" : !this.player ? "Well it seems I cant find the cable for my 🎧" : this.queue.length === 0 ? "I have no records to play so please provide some" : null;
+        if (check) throw check;
+
+        const song = this.queue[0];
+        this.player.play(song.track);
+        this.playing = true;
+        return this.player;
+    }
+
+    /**
+	 * Pauses the player and returns this
+     * @since 2.0.1
+	 * @returns {this}
+	 */
+    pause() {
+        if (!this.player) return null;
+        this.player.pause();
+        this.paused = true;
+        return this;
+    }
+
+    /**
+	 * Resumes the player and returns this
+     * @since 2.0.1
+	 * @returns {this}
+	 */
+    resume() {
+        if (!this.player) return null;
+        this.player.resume();
+        this.paused = false;
+        return this;
+    }
+
+    /**
+	 * Depending on the current state it will either pause or resume
+     * @since 2.0.1
+	 * @returns {this}
+	 */
+    smartPlay() {
```

Comment:
This seems pointless, should be in play command

Reply:
Well the naming is wrong. It is mainly smart pause